app/models/vacancy.rb

```diff
@@ -321,6 +324,13 @@ def convert_date_to_unix_time(date)
     Time.zone.at(date.to_time).to_datetime.midday.to_i
   end
 
+  def convert_time_to_unix_time(time)
+    # nil.to_i returns 0 in unix time (midnight 1970), so if time is nil we should return nil.
+    return nil if time == nil
```

Comment:
Advice here: 

It's always recommended not to return `nil` directly. If you're returning `nil` there's probably something else higher up that we're missing.

In this case, we want to somehow say that vacancies might not have an expiry time.
In that case, I'd set a default value on the attribute `expiry_time` to be midnight of the expiry date.

Reply:
Vacancies might not have an expiry date either (a vacancy is created in the database, and could be indexed, before all the required data to list a vacancy has been entered by a user)

Reply:
I believe that's a DateTime object, @ddippolito so combining our two comments it could just be:

```
time.nil? ? DateTime.now.to_i : time.to_i
```

That said, Alogolia needs in seconds or milliseconds since the epoch for sorting purposes, so zero would be fine under the circumstances.

Reply:
@cpjmcquillan is right, but we could still account for the case where we do have expiry_date but no expiry_time.

Reply:
Might be worth renaming that attribute, @david-mears-dfe to something like `expiry_time_epoch` or suchlike and sending the 'real' attribute as a string. That way the record can be sorted by time and the client doesn't need to do anything to covert the sorting timestamp.

Reply:
@tatyree I thought sorting was achieved through replicas of the index?

Reply:
@cpjmcquillan I think we're talking at cross purposes here. 

"Note that this attribute needs to be a numeric value for Algolia to be able to sort on it."

From

https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/how-to/sort-an-index-by-date/#configuring-the-index